### ISSUE TITLE
504 when merge comments contain certain characters

### ISSUE THREAD

**Issue by Freso** [internetarchive/openlibrary#9880]
### TL;DR

`’` and possibly other Unicode/non-ASCII characters in merge comments cause merges to fail with a 504.

### Description

When attempting to do [this author merge](https://openlibrary.org/recentchanges/2024/09/16/merge-authors/138022304), I tried to add a helpful merge comment… which made the request error out with a 504 error message.

First attempt:
> "A. I. Bakunin" was created for https://openlibrary.org/books/OL20549470M - so seems like some misspelling/typo/OCR-error’d of Bakunin

Second attempt:
> “A. I. Bakunin” was created for https://openlibrary.org/books/OL20549470M - so seems like some misspelling/typo/OCR-error’d of Bakunin

Third attempt:
> “A. I. Bakunin” was created for https://openlibrary.org/books/OL20549470M so seems like some misspelling/typo/OCR-error’d of Bakunin

Fourth attempt:
> “A. I. Bakunin” was created for /books/OL20549470M so seems like some misspelling/typo/OCR-error’d of Bakunin

Fifth (and successful) attempt:
> "A. I. Bakunin" was created for https://openlibrary.org/books/OL20549470M so seems like some misspelling/typo/OCR-error'd of Bakunin

The only difference between 3rd and 5th attempt is the use of `“”’` vs. `""'` and given that the 1st attempt only had `""` this means that at least `‘` is causing the merge to fail. Whether `“”` also cause issues, I don’t know. Remains to be seen for testing or future merge comments. :)

### Reproducing the bug

1. Go to a merge page
2. Add a merge comment with some non-ASCII characters (or at least `‘`)

* Expected behavior: Merge goes through.
* Actual behavior: 504 error and merge doesn’t happen.

### Context

- Browser (Chrome, Safari, Firefox, etc): Firefox 130.0
- OS (Windows, Mac, etc): Linux (I use Arch btw™)
- Logged in (Y/N): Y
- Environment (prod, dev, local): prod

#### Requirements Checklist
* [ ]

#### Related files

*

#### Stakeholders

*
<hr>

#### Instructions for Contributors

- Please [run these commands](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#working-on-your-branch) to ensure your repository is up to date **before** [creating a new branch](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) to work on this issue and **each time after** pushing code to Github, because the pre-commit bot may add commits to your PRs upstream.

**Comment by cdrini** [comment]
Sending this one to you @jimchamp since you're more familiar with this layer of the Merge Queue

**Comment by jimchamp** [comment]
@Freso, I've copied and pasted your first and fourth attempts, and was successfully able to POST both as comments on merge request #164788:

![image](https://github.com/user-attachments/assets/f3fc904b-96ac-42f3-998d-f3311a70c34b)

I'm assuming that there were some capacity issue that caused timeouts on the day that you tried to leave these comments.

**Comment by jimchamp** [comment]
Special characters seem to be working:

![image](https://github.com/user-attachments/assets/4c94115e-b2f0-4b1e-b66f-e1d7f4a87a59)
